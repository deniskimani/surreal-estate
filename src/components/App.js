/* eslint-disable import/no-duplicates */
import { Route, Routes } from "react-router-dom";
import "../styles/app.css";
import { library } from "@fortawesome/fontawesome-svg-core";
import { faB, faR } from "@fortawesome/free-solid-svg-icons";
import {
  faCoffee,
  faBed,
  faShower,
  faBath,
  faSterlingSign,
  faEnvelope,
} from "@fortawesome/free-solid-svg-icons";

import NavBar from "./NavBar";
import Properties from "./Properties";
import AddProperties from "./AddProperties";

library.add(
  faB,
  faR,
  faCoffee,
  faBed,
  faBath,
  faShower,
  faSterlingSign,
  faEnvelope
);

const App = () => {
  return (
    <div className="App">
      <NavBar />
      <Routes>
        <Route element={<Properties />} path="/" />
        <Route element={<AddProperties />} path="/add-property" />
      </Routes>
    </div>
  );
};

export default App;
