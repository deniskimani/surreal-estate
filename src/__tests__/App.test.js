import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import App from "../components/App";

describe("", () => {
  const { asFragment } = render(
    <BrowserRouter>
      <App />
    </BrowserRouter>
  );

  it("renders correctly", () => {
    expect(asFragment()).toMatchSnapshot();
  });
});
