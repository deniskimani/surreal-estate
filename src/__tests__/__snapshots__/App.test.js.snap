// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` renders correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="nav-bar"
    >
      <a
        class="logo"
        href="/"
      >
        <img
          alt="logo"
          src="https://mcrcodes.s3.eu-west-2.amazonaws.com/course/surreal-estate/logo.png"
        />
      </a>
      <ul
        class="links"
      >
        <li>
          <a
            class="item"
            href="/"
          >
            View Properties
          </a>
        </li>
        <li>
          <a
            class="item"
            href="/add-property"
          >
            Add Property
          </a>
        </li>
      </ul>
    </div>
    <div>
      <h3>
        Properties
      </h3>
    </div>
  </div>
</DocumentFragment>
`;
